File: codelens_ai/frontend/src/types/improvement.ts
// frontend/src/types/improvement.ts

export type ImprovementItem = {
  title: string;
  category: string;
  priority: "High" | "Medium" | "Low" | string;
  description: string;
  rationale: string;
};

export type ImprovementListResponse = {
  repo_id: number;
  total: number;
  improvements: ImprovementItem[];
};

File: codelens_ai/backend/app/schemas/improvement.py
# backend/app/schemas/improvement.py

from pydantic import BaseModel


class ImprovementItem(BaseModel):
    title: str
    category: str
    priority: str
    description: str
    rationale: str


class ImprovementListResponse(BaseModel):
    repo_id: int
    total: int
    improvements: list[ImprovementItem]